apps/typegpu-docs/src/content/docs/tooling/unplugin-typegpu.mdx

```diff
@@ -0,0 +1,144 @@
+---
+title: Build Plugin
+description: A guide on how to use the optional build plugin for TypeGPU
+---
+
+[unplugin-typegpu](https://www.npmjs.com/package/unplugin-typegpu) is a tool for TypeGPU applications,
+which automatically transforms the code during build time,
+unlocking some new features, optimizations and quality-of-life perks for the developers.
+While using the plugin is not required in majority of scenarios, it is highly recommended when writing GPU programs using TypeGPU.
```

Comment:
Let me know what you think about this change. I think it's a bit more straightforward and terse.
```suggestion
[unplugin-typegpu](https://www.npmjs.com/package/unplugin-typegpu) is an optional (but highly recommended) tool for projects using TypeGPU. It hooks into your bundler of choice, and unlocks new features, optimizations and quality-of-life improvements.
```